contracts/exchange/contracts/src/MixinWrapperFunctions.sol

```diff
@@ -243,13 +242,22 @@ contract MixinWrapperFunctions is
                 remainingMakerAssetFillAmount
             );
 
+            // The `makerAssetData` must be the same for each order.
+            // Rather than checking equality, we point the `makerAssetData` of each order to the same memory location.
+            // This is less expensive than checking equality.
+            bytes memory originalMakerAssetData = orders[i].makerAssetData;
+            orders[i].makerAssetData = makerAssetData;
+
             // Attempt to sell the remaining amount of takerAsset
             LibFillResults.FillResults memory singleFillResults = fillOrderNoThrow(
                 orders[i],
                 remainingTakerAssetFillAmount,
                 signatures[i]
             );
 
+            // Restore the original `makerAssetData` so we're non-destructive.
+            orders[i].makerAssetData = originalMakerAssetData;
```

Comment:
Hmm, this change doesn't feel super necessary to me since this isn't intended to be a library or used in external contracts. We also do something similar in `matchOrders`, right? Is there anywhere where we actually reuse the in-memory orders after calling these functions?

Reply:
It was to preserve the order hashes in the event of a rich revert, but we won't need it anymore if we aren't returning order hashes.